server/service/src/sync/sync_serde.rs

```diff
@@ -1,11 +1,27 @@
 use chrono::{Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime};
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::{
+    de::{value::StrDeserializer, IntoDeserializer},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
 
-pub fn empty_str_as_option<'de, D: Deserializer<'de>>(d: D) -> Result<Option<String>, D::Error> {
+pub fn empty_str_as_option_string<'de, D: Deserializer<'de>>(
+    d: D,
+) -> Result<Option<String>, D::Error> {
     let s: Option<String> = Option::deserialize(d)?;
     Ok(s.filter(|s| !s.is_empty()))
 }
 
+pub fn empty_str_as_option_generic<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
```

Comment:
Just call it `empty_str_as_option`?  appending generic sounds a bit weird...

Reply:
I do agree it's a bit weird, but wanted to differentiate from the other one. Happy to trim!

Just a note, the `empty_str_as_option_string` is faster if genuinely just looking for a string.

Reply:
think its already differentiated enough because the other one is called `empty_str_as_option_string`

Reply:
e8c8fd751ff21090229157626296aef97df46a97